pkg/sql/colexec/insert/insert.go

```diff
@@ -183,3 +194,236 @@ func fillRow(tmpBat *batch.Batch,
 	vec.Free(proc.Mp())
 	return err
 }
+
+func resetInsertBatchForOnduplicateKey(proc *process.Process, originBatch *batch.Batch, insertArg *Argument) (*batch.Batch, error) {
```

Comment:
you should make it a new colexec operator to do this reset work.

I think insert operator should be a pure insert.

Reply:
done